react-native/app/index.tsx

```diff
@@ -9,38 +10,47 @@ import {
   View,
 } from "react-native";
 import { createClient, Code, ConnectError } from "@connectrpc/connect";
-import { createXHRGrpcWebTransport } from "./custom-transport";
+import { createConnectTransport } from "@connectrpc/connect-web";
 import {
   ElizaService,
   IntroduceRequestSchema,
 } from "../gen/connectrpc/eliza/v1/eliza_pb.js";
-// Needed to polyfill TextEncoder/ TextDecoder
-import "fast-text-encoding";
 import { create } from "@bufbuild/protobuf";
-import { polyfills } from "./polyfills";
 
-polyfills();
-
-interface Response {
+interface ConvoResponse {
   text: string;
   sender: "eliza" | "user";
 }
 
 function Index() {
   const [statement, setStatement] = useState<string>("");
   const [introFinished, setIntroFinished] = useState<boolean>(false);
-  const [responses, setResponses] = useState<Response[]>([
+  const [responses, setResponses] = useState<ConvoResponse[]>([
     {
       text: "What is your name?",
       sender: "eliza",
     },
   ]);
 
-  // Make the Eliza Service client
+  // Make the Eliza Service client using grpc-web transport.
   const client = createClient(
     ElizaService,
-    createXHRGrpcWebTransport({
+    createConnectTransport({
       baseUrl: "https://demo.connectrpc.com",
+      // Customize fetch with the Expo fetch implementation
+      fetch: (input, init) => {
+        if (typeof input !== "string") {
+          throw new Error(
+            "expo/fetch requires the first argument to be a string URL",
+          );
+        }
+        return fetch(input, {
+          ...init,
+          body: init?.body ?? undefined,
+          credentials: init?.credentials ?? undefined,
+          signal: init?.signal ?? undefined,
+        }) as unknown as Promise<Response>;
```

Comment:
Just realized - isn't this the same?

```suggestion
        return fetch(input, init) as unknown as Promise<Response>;
```

If this raises a type error, it'll most likely be fine to cast as RequestInit, similar to the response.